snuba/web/query.py

```diff
@@ -185,7 +186,13 @@ def _run_query_pipeline(
         metrics.increment("sample_without_turbo", tags={"referrer": request.referrer})
 
     if request.settings.get_dry_run():
-        query_runner = _dry_run_query_runner
+        storage_name = "<multiple>"
+        if isinstance(request.query, LogicalQuery):
+            entity = get_entity(request.query.get_from_clause().key)
+            storage = entity.get_all_storages()[0]
+            assert isinstance(storage, ReadableTableStorage)
+            storage_name = storage.get_storage_key().value
```

Comment:
@lynnagara 
Are you talking about line 192 here? From most entity definitions I've seen, we have the form:
```python3
super().init(storages=[some_selected_storage], ...)
```

So when we do `entity.get_all_storages()[0]`, it just returns that one right?

Eg: https://github.com/getsentry/snuba/blob/de14212c8f504144359fa63692a261c796d41067/snuba/datasets/entities/events.py#L198-L199

How different is this than how a regular Snuba query is processed, the entity returns the appropriate storage from its list of storages no?

The only deviations I've seen from this is in the metrics, outcomes, and sessions entities and even those just have two storages instead of one, with the first one being good enough to query from as the storage.

I could change it to be two drop downs, one for dataset and one for storage but I think the storage dropdown would contain exactly 1 or 2 storages to choose from (read/writable + maybe materialized).

Reply:
AFAIK it's quite different to how a regular query is processed which goes through the StorageSelector. I don't think we should rely on the fact that today our storages happen to be just one or two within each entity, because we'd want it to still work with future datasets/entities where that might not be the case.

To be honest, I personally don't think the SnQL tracing tool adds all that much additional value once we have the SnQL -> SQL converter, since we can just grab the output of that and stick it into the existing tool. Especially if combining the two functionalities forces us to make these kinds of assumptions about storages within an entity.

Reply:
I'm all for deleting code, so I'll remove that section in favor of the SnQL to SQL translate -> SQL Tracing workflow.